docs/components/doc-management.md

```diff
@@ -1 +1,38 @@
-TBD
+This page details document management, including the ingestion and parsing of files, as well as searching through the files present in the Database.
+
+### Document Ingestion
+PDF files are parsed on a per-document, per-page basis. Once broken down into chunks (each of which is a page), a locally hosted model - `Alibaba-NLP/gte-base-en-v1.5` by default - generates vector embeddings for each chunk page. Each page/ chunk, with its metadata and embedding, is stored in the PostgreSQL database using vector indexing for efficient similarity search.
+
+```mermaid
+sequenceDiagram
+    autonumber
+    User->>API: Upload file for ingestion
+    API->>DocumentService: Parse file into pages (chunks)
+    DocumentService-->>API: Return page-based chunks
+    API->>DocumentService: Generate embeddings for pages
+    DocumentService-->>API: Return embeddings
+    API->>Db: Save page chunks and embeddings
+    Db-->>API: Acknowledge save
+    API-->>User: Return file ID
+```
+
+#### Tip: All ingested documents can be easily viewed with their metadata by querying the `list_docs` `GET` endpoint.
+
+### Search for similar documents
+
+The search flow looks as follows:
+
+```mermaid
+sequenceDiagram
+    autonumber
+    User->>API: Enter search query for similar documents
+    API->>VectorDB: Embed query and compare closest chunks to query
```

Comment:
You can also have self loops. So maybe a self loop for embed query? or call LLM for it. And then to the VectorDB to get closest. I should also make this change to mine.